### PR TITLE
feat: Better Scrobbler Plugin

### LINKED ISSUE #1350: [Feature Request]: add more scrobble options
### Preflight Checklist

- [X] I use the latest version of YouTube Music (Application).
- [X] I have searched the [issue tracker](https://github.com/th-ch/youtube-music/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

since not everyone uses lastfm, it would be nice to have alternative scrobble platform added too like listenbrainz or librefm

### Proposed Solution

add listenbrainz or librefm as an alternative to lastfm

### Alternatives Considered

or maybe you could integrate this chrome extension into an app

https://github.com/web-scrobbler/web-scrobbler

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Serial-ATA**
This converts the `last-fm` plugin into the new `scrobbler` plugin.

This currently supports Last.fm and ListenBrainz.

All translation strings for the `last-fm` plugin had to be removed.

Both services have their own configuration windows:

image

image

image

closes #1350

### CODE REVIEW

**Comment by JellyBrick** [comment]
It needs a migration script.

see https://github.com/th-ch/youtube-music/blob/master/src/config/store.ts

File: src/plugins/scrobbler/index.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { createPlugin } from '@/utils';
+import { t } from '@/i18n';
+import { onMenu } from './menu';
+import { onConfigChange, onMainLoad } from './main';
+
+export interface ScrobblerPluginConfig {
+  enabled: boolean,
+  lastfm_options: {
```
Comment (JellyBrick) [inline src/plugins/scrobbler/index.ts]: How about this?

```typescript
scrobblers: {
  lastfm: { ... },
  listenbrainz: { ... },
},
```

**Review by JellyBrick — CHANGES_REQUESTED** [review]

File: src/plugins/scrobbler/index.ts (hunk 1 above)
Comment (Serial-ATA) [reply src/plugins/scrobbler/index.ts]: Good call

**Comment by Serial-ATA** [comment]
> It needs a migration script.

Is there a way to test migrations? I made one, but don't know how to try it out.

File: src/config/store.ts [hunk 2]
```diff
@@ -6,6 +6,39 @@ import defaults from './defaults';
...
+
+      for (const key of lastFmKeys) {
+        const value = store.get('plugins.lastfm.' + key) as string | undefined;
+        if (value !== undefined) {
+          scrobblerConfig.scrobblers.lastfm[key] = value;
+        }
+      }
+
+      store.set('plugins.scrobbler', scrobblerConfig);
+    }
```
Comment (JellyBrick) [inline src/config/store.ts]:
```suggestion
    const lastfmConfig = store.get('plugins.lastfm') as {
      enabled?: boolean;
      token?: string;
      session_key?: string;
      api_root?: string;
      api_key?: string;
      secret?: string;
    };
    if (lastfmConfig) {
      const scrobblerConfig = store.get(
        'plugins.scrobbler',
      ) as {
        enabled?: boolean;
        scrobblers: {
          lastfm: {
            enabled?: boolean;
            token?: string;
            session_key?: string;
            api_root?: string;
            api_key?: string;
            secret?: string;
          };
        };
      };

      if (lastfmConfig) {
        scrobblerConfig.enabled = lastfmConfig.enabled;
        scrobblerConfig.scrobblers.lastfm = lastfmConfig;
        store.set('plugins.scrobbler', scrobblerConfig);
      }
    }
```

File: src/config/store.ts [hunk 3]
```diff
@@ -6,6 +6,39 @@ import defaults from './defaults';
 import { DefaultPresetList, type Preset } from '@/plugins/downloader/types';
 
 const migrations = {
+  '>=3.2.2'(store: Conf<Record<string, unknown>>) {
```
Comment (JellyBrick) [inline src/config/store.ts]:
```suggestion
  '>=3.3.0'(store: Conf<Record<string, unknown>>) {
```
If the plugin is added, we must bump the minor version.

**Review by JellyBrick — CHANGES_REQUESTED** [review]

File: src/config/store.ts (hunk 2 above)
Comment (Serial-ATA) [reply src/config/store.ts]: Thanks, I got that completely backwards.

**Review by JellyBrick — APPROVED** [review]

**Pull request merged**